src/components/Todo/Todo.js

```diff
@@ -1,8 +1,65 @@
-import React from 'react'
+import React, { useState } from 'react'
 import ListItem from 'components/ListItem/ListItem'
+import Input from 'components/Input/Input'
+import Button from 'components/Button/Button'
+import { FaEdit } from 'react-icons/fa'
+import { MdCancel } from 'react-icons/md'
 
-const Todo = ({ children, ...props }) => {
-  return <ListItem {...props}>{children}</ListItem>
+const Todo = ({ onUpdateText, id, value, children, ...props }) => {
+  const [editing, setEditing] = useState(false)
+  const [toDoText, setToDoText] = useState(value)
+  const [changeElement, setChangeElement] = useState(false)
+
+  const showEditIcon = () => setEditing(true)
+  const hideEditIcon = () => setEditing(false)
+
+  const editToDo = () => setChangeElement(true)
+
+  const changeToDoText = (event) => setToDoText(event.target.value)
+
+  const saveChangesEnter = (event) => {
+    if (event.key === 'Enter' && toDoText.length > 0) {
+      onUpdateText(id, toDoText)
+      setChangeElement(false)
+    }
+  }
+
+  const escapeEdit = (event) => {
+    if (event.key === 'Escape') {
+      setToDoText(value)
+      setChangeElement(false)
```

Comment:
Acá podés evitar duplicar código llamando a `cancelEdit`

```suggestion
      cancelEdit()
```

Reply:
Pablo, estoy tratando de aplicar este cambio, pero no me funciona. ¿Me faltará algún parámetro?

Reply:
Lo había puesto en otro lado (palmface). Puede fallar.

Reply:
Algo toqué igual y ahora no anda buaaaa; estoy tratando de resolverlo pero no sé.